sgkit/stats/ld.py

```diff
@@ -61,3 +70,377 @@ def rogers_huff_r_between(gn0: ArrayLike, gn1: ArrayLike) -> float:  # pragma: n
 @njit(nogil=True, fastmath=True, cache=True)  # type: ignore
 def rogers_huff_r2_between(gn0: ArrayLike, gn1: ArrayLike) -> float:  # pragma: no cover
     return rogers_huff_r_between(gn0, gn1) ** 2  # type: ignore
+
+
+def ld_matrix(
+    ds: Dataset,
+    *,
+    dosage: Hashable = variables.dosage,
+    threshold: Optional[float] = None,
+    ld_score: Optional[Hashable] = None,
+) -> DataFrame:
+    """Compute a sparse linkage disequilibrium (LD) matrix.
+
+    This method computes the Rogers Huff R2 value for each pair of variants in
+    a window, and returns those that exceed the provided threshold, as a sparse
+    matrix dataframe.
+
+    Parameters
+    ----------
+    ds
+        Dataset containing genotype dosages.
```

Comment:
It would be nice IMO if the windowing requirement and how to add them was elevated to somewhere like this in the docstrong rather than being implicit in "Raises".

Reply:
Done